src/frontend/README.md

```diff
@@ -1,7 +1,128 @@
-# frontend
+# frontend service
 
-Run the following command to restore dependencies to `vendor/` directory:
+The **frontend** service is responsible for rendering the UI for the store's website.
+It serves as the main entry point for the application routing requests to their
+appropriate backend service.
+The application uses Server Side Rendering (SSR) to generate HTML consumed by
+the browser.
 
-```sh
-dep ensure --vendor-only
+## OpenTelemetry features
+
+| Auto-Instrumented tracing | Custom spans | Custom span attributes | Metrics | Logs |
+|---------------------------|--------------|------------------------|---------|------|
+| X                         |              | X                      |         |      |
+
+## OpenTelemetry instrumentation
+
+### Initialization
+
+The OpenTelemetry SDK is initialized in `main` using the `InitTraceProvider` function.
+
+```go
+func InitTracerProvider() *sdktrace.TracerProvider {
+    ctx := context.Background()
+
+    exporter, err := otlptracegrpc.New(ctx)
+    if err != nil {
+        log.Fatal(err)
+    }
+    tp := sdktrace.NewTracerProvider(
+        sdktrace.WithSampler(sdktrace.AlwaysSample()),
+        sdktrace.WithBatcher(exporter),
+    )
+    otel.SetTracerProvider(tp)
+    otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}))
```

Comment:
Is this required to be explicitly set, or is that the default?

Reply:
The Go SDK does not have a default propagator, so this is required.